presentation-api/receiving-ua/support/PresentationReceiver_create_receiving-ua.html

```diff
@@ -106,27 +106,30 @@
         let db;
         const checkIndexedDB = () => {
             if ('indexedDB' in window) {
-                let req = indexedDB.open(dbName, 1), store;
-                let eventWatcher = new EventWatcher(t, req, ['upgradeneeded', 'success']);
-                return eventWatcher.wait_for('upgradeneeded').then(() => {
-                    db = req.result;
-                    const store = db.createObjectStore(storeName, { keyPath: 'id' });
-                    return eventWatcher.wait_for('success');
-                }).then(() => {
-                    db = req.result;
-                    const transaction = db.transaction(storeName, 'readwrite');
-                    store = transaction.objectStore(storeName);
-                    req = store.openCursor();
-                    eventWatcher = new EventWatcher(t, req, 'success');
-                    return eventWatcher.wait_for('success');
-                }).then(() => {
-                    assert_equals(req.result, null, 'Indexed Database is set to an empty storage.');
-                    req = store.put(storeData.parent);
-                    eventWatcher = new EventWatcher(t, req, 'success');
-                    return eventWatcher.wait_for('success');
-                }).then(() => {
-                    db.close();
-                    db = null;
+                let req = indexedDB.open(dbName, 1), store, transaction;
+                return new Promise((resolve, reject) => {
+                    // The test would fail when the database is already created by the controlling UA
```

Comment:
According to the IndexedDB spec, you can test for existence of a database by a non-empty version property on the result object.

https://developer.mozilla.org/en-US/docs/Web/API/IDBDatabase

Perhaps the test would be simpler by just checking whether a database already exists (thus indicating whether the database was leaked across receiving browsing contexts).

Reply:
That's a good idea. Regarding IndexedDB, we can simply check version of a database. I'll simplify and submit these codes later.

BTW, I'm afraid that the MDN article might be incorrect. With regard to `indexedDB.open`, the [IndexedDB spec](https://www.w3.org/TR/IndexedDB/#dom-idbfactory-open) says:

>If version is not given and no database with that name exists, a new database will be created with version equal to 1.

Anyway, we can still check whether the database has been previously created or not by looking at version of the database.